docs/rails.md

```diff
@@ -1,19 +1,25 @@
-# Rails conventions
+# Rails-specific best practices
 
-## App behavior should be configured via environment variables
+## Application configuration
 
-When implementing new features, it should not depend on current
-`RAILS_ENV` value, but environment variable(s) specific to the
-feature.
+Application configuration should **depend on environment variables**,
+and absolutely **not rely on `RAILS_ENV`** value (nor `Rails.env`),
+respecting the [12-factor app methodology][12-factor config].
```

Comment:
I agree that the original sub-title and intro wasn't expressive enough.

I wanted to make a distinction between "business" features/behavior, and
low-level, framework stuff. It wasn't clear enough, but with current change
it's totally gone.

I also think that the original text was more informative.

New text also has a small contradiction: `RAILS_ENV`  *is* an environment
variable.